app/reducers/index.js

```diff
@@ -46,6 +50,28 @@ const redatApp = (state = defaultState, action) => {
     case 'REMOVE_DAT':
       const { [action.key]: del, ...dats } = state.dats
       return { ...state, dats }
+    case 'INSPECT_DAT':
+      return { ...state,
+        inspect: { ...state.inspect,
+          key: action.key,
+          show: true
+        }
+      }
+    case 'INSPECT_DAT_CLOSE': 
+      return { ...state, 
+        inspect: { ...state.inspect,
+          key: null,
+          show: false
+        }
+      }
+    case 'ADD_FILE_TO_DAT':
+      return { ...state,
+        dats: { ...state.dats,
+          [action.key]: {...state.dats[action.key],
+            files: action.files
```

Comment:
wording is misleading here: the event name suggests a single file is added, however looking at the code it's overwriting the whole files array. to be consistent with other events, it should be renamed to `DAT_FILES`